### PR TITLE
fix(auth): handle firestore play fetch error

### LINKED ISSUE #11026: WError: An internal server error occurred
https://sentry.prod.mozaws.net/operations/fxa-auth-prod/issues/14943344/?referrer=github_plugin

```
WError: An internal server error occurred
  File "/fxa/packages/fxa-auth-server/dist/fxa-auth-server/lib/payments/google-play/user-manager.js", line 72, in UserManager.queryCurrentSubscriptions
    const libraryError = new Error(err.message);
  ?, in runMicrotasks
  File "/fxa/packages/fxa-auth-server/dist/fxa-auth-server/lib/payments/capability.js", line 302, in async CapabilityService.fetchSubscribedProductsFromPlay
    const allPurchases = await this.playBilling.userManager.queryCurrentSubscriptions(uid);
  ?, in async Promise.all
  File "/fxa/packages/fxa-auth-server/dist/fxa-auth-server/lib/payments/capability.js", line 178, in async CapabilityService.subscribedProductIds
    const [subscribedStripeProducts, subscribedPlayProducts] = await Promise.all([
...
(8 additional frame(s) were not displayed)
```

This error should be caught during capability lookup, and the purchase the token corresponds to not considered to be active in the capability response.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXA-4203)

### PR DESCRIPTION
**Pull request by bbangert**
Because:

* We want to gracefully handle firestore fetch errors that we can't
  remedy.

This commit:

* Catches the firestore fetch error and gracefully continues.

Closes #11026

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: packages/fxa-auth-server/lib/payments/capability.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ import { PlayBilling } from './google-play/play-billing';
 import { SubscriptionPurchase } from './google-play/subscription-purchase';
 import { StripeHelper } from './stripe';
 import error from '../error';
+import { PurchaseQueryError } from './google-play/types';
```
Comment (biancadanforth) [inline packages/fxa-auth-server/lib/payments/capability.ts:17]: What are some possible root causes for this error? Is there anything in our control?

**Review by biancadanforth — APPROVED** [review]
Hey Ben; thanks for the PR. This does now catch the error, though IIUC the problem of possibly denying access to an active Google IAP subscriber persists? We would now return a 200 though for the GET account/profile instead of a 500.

Edit: Assuming CI passes of course -- just noticed there were some failures.

File: packages/fxa-auth-server/lib/payments/capability.ts (hunk 1 above)
Comment (bbangert) [reply packages/fxa-auth-server/lib/payments/capability.ts:17]: Nope, this one is purely from Firestore.

File: packages/fxa-auth-server/lib/payments/capability.ts [hunk 2]
```diff
@@ -345,14 +346,24 @@ export class CapabilityService {
...
+        ? []
+        : this.stripeHelper.purchasesToProductIds(purchases);
+    } catch (err) {
+      if (err.name === PurchaseQueryError.OTHER_ERROR) {
+        this.log.error('Failed to query purchases from Google Play', {
+          uid,
+          err,
+        });
+      }
+      return [];
```
[the remark this answers is not in the document]
Comment (bbangert) [reply packages/fxa-auth-server/lib/payments/capability.ts:365]: Currently it goes to Sentry, but its Firestore throwing an error, there's nothing we can actually do. In the several months its been running, we've only had it happen during deploys, and I believe this one overlapped with when we were missing a Firestore group collection index.

**Pull request merged**